cclib/bridge/cclib2horton.py

```diff
@@ -0,0 +1,145 @@
+    attributes = {}
+    
+    if (hortonver == 2):
+        # if coordinates are known:
+        if (isinstance(atomcoords, numpy.ndarray)):
+            attributes['coordinates'] = atomcoords[-1]
+        # if atomic numbers are known:
+        if (isinstance(atomnos, numpy.ndarray)):
+            attributes['numbers'] = atomnos,
+        # if orbital coefficients known:
+        if (isinstance(mocoeffs, numpy.ndarray)):
+            attributes['orb_alpha'] = mocoeffs[0]
+            # and if unrestricted:
+            if (len(mocoeffs) == 2):
+                attributes['orb_beta'] = mocoeffs[1]
+        # if multiplicity known:
+        if (mult != None):
+            attributes['ms2'] = mult
+        # if pseudopotentials exist (type-checked by horton):
+        if (isinstance(coreelectrons, numpy.ndarray)):
+            attributes['pseudo_numbers'] = coreelectrons
+        # if mulliken charges are known:
+        if (isinstance(atomcharges, dict) and 'mulliken' in atomcharges):
+            attributes['mulliken_charges'] = atomcharges['mulliken']
+        # if npa charges are known:
+        if (isinstance(atomcharges, dict) and 'natural' in atomcharges):
+            attributes['npa_charges'] = atomcharges['natural']
+        # if polarizability is known (type-checked by horton):
+        if (isinstance(polarizability, numpy.ndarray)):
+            attributes['polar'] = polarizability
+            
+    elif (hortonver == 3):
+        # In horton 3 IOData, inputs are type-checked
+        # if coordinates are known; numpy.ndarray (size natom-by-3)
+        if (isinstance(atomcoords, numpy.ndarray)):
+            attributes['atcoords'] = atomcoords[-1]
+        # if atomic numbers are known; numpy.ndarray (size natom)
+        if (isinstance(atomnos, numpy.ndarray)):
+            attributes['atnums'] = atomnos,
+        # if orbital coefficients known; iodata.orbitals.MolecularOrbitals
+        if (isinstance(mocoeffs, numpy.ndarray)):
+            attributes['mo'] = MolecularOrbitals(kind = 'restricted', norba = len(mocoeffs[0]), coeffs = mocoeffs[0].T)
+            # and if unrestricted:
+            if (len(mocoeffs) == 2):
+                attributes['mo'].kind = 'unrestricted'
+                attributes['mo'].norbb = len(mocoeffs[1])
+                attributes['mo'].coeffs.append(mocoeffs[1].T)
+        # if multiplicity known; float / should not be set when mocoeffs present
+        if (mult != None and not isinstance(mocoeffs, numpy.ndarray)):
+            attributes['spinpol'] = mult - 1 # horton has 2S+1, iodata has 2S
+        # if pseudopotentials exist; numpy.ndarray (size natom)
+        if (isinstance(coreelectrons, numpy.ndarray)):
+            attributes['atcorenums'] = coreelectrons
+        # if mulliken charges are known; dict of numpy.ndarrays (size natom)
+        if (isinstance(atomcharges, dict)):
+            attributes['atcharges'] = atomcharges
```

Comment:
Can you add an `else:` that raises an error?

Reply:
As far as I know, `atomcharges` are not always present in input data? The if statement is supposed to detect whether any valid information was passed on or not (since default value for `atomcharges` is `None`). I see that this may not be the cleanest way to do this -- let me know I could modify this further. Thanks!

Reply:
This would be more natural if we passed in the ccData object.